src/Card/index.js

```diff
@@ -1,50 +1,107 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './style.css';
 import { BiEdit } from 'react-icons/bi';
 import { AiOutlineSave } from 'react-icons/ai';
 import { ImCancelCircle } from 'react-icons/im'
 
-const card = props => {
+const Card = props => {
+  const [cardState, setCardState] = useState({
+    checked: false,
+    isEditable: false,
+    data: {
+      headerData: props.headerData,
+      bodyData: props.bodyData
+    },
+    tempData: {
+      headerData: props.headerData,
+      bodyData: props.bodyData
+    }
+  });
+
+  const changeStyleHandler = () => {
+    setCardState({
+      ...cardState,
+      checked: !cardState.checked
+    })
+  };
+
+  const openEditModeHandler = () => {
+    setCardState({
+      ...cardState,
+      isEditable: !cardState.isEditable,
+      checked: false
+    })
+  };
+
+  const cardDataChangeHandler = (event, prop) => {
+    setCardState({
+      ...cardState,
+      tempData: {
+        ...cardState.tempData,
+        [prop]: event.target.value
+      }
+    })
+  };
+
+  const saveChangesHandler = () => {
+    setCardState({
+      ...cardState,
+      isEditable: false,
+      data: {
+        ...cardState.tempData
+      }
+    })
+  };
+
+  const cancelChangesHandler = () => {
+    setCardState({
+      ...cardState,
+      isEditable: false,
+      tempData: {
+        ...cardState.data
+      }
+    })
+  };
+
   const renderEditMode = () => {
     return (
       <div className='header-edit-mode-buttons' >
-          <AiOutlineSave className='save-button' onClick={props.onSaveChanges}/>
-          <ImCancelCircle className='cancel-button' onClick={props.onCancelChanges}/>
+          <AiOutlineSave className='save-button' onClick={saveChangesHandler}/>
+          <ImCancelCircle className='cancel-button' onClick={cancelChangesHandler}/>
        </div>
       )
   };
 
   const renderReadMode = () => {
     return (
         <div className='header-default-buttons'>
-          <BiEdit className='edit-button' onClick={props.openEditMode}/>
-          <input type='checkbox' onChange={props.onChange} />
+          <BiEdit className='edit-button' onClick={openEditModeHandler}/>
+          <input type='checkbox' onChange={changeStyleHandler} />
         </div>
       )
   };
 
   return (
-    <div className={props.checked ? 'card active-status' : 'card'}>
-        <div className={props.checked ? 'card-header active' : 'card-header'}>
+    <div className={cardState.checked ? 'card active-status' : 'card'}>
+        <div className={cardState.checked ? 'card-header active' : 'card-header'}>
           <input type="text"
             className='card-header-text' 
-            readOnly={!props.isEditable}
-            value={props.headerText}
-            onChange={props.headerTextChangeHandler}
+            readOnly={!cardState.isEditable}
+            value={cardState.tempData.headerData}
+            onChange={event => cardDataChangeHandler(event, 'headerData')}
```

Comment:
As alternative, you can pass custom property in input and get its value as `event,target.<customProperty>` in the handler, in the same way as you achieve` value`.